### PR TITLE
Migrate instanceUpdate on service.tsx

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Migrate instanceUpdate on service.tsx

image

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 3ebf5d7</samp>

This pull request refactors the dashboard service to use the public API for workspace status, and adds test cases for converting strings to and from the `Phase` enum in the public API. It also removes some unused code and adds a TODO comment.

</details>

## Related Issue(s)

Fixes #

## How to test

- Workspace create stop and restart should work like before
- Open another tab for the same workspace and stop / restart, both tabs should be updated

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-papi-wa87eb04d5c1</li>
	<li><b>🔗 URL</b> - <a href="https://hw-papi-wa87eb04d5c1.preview.gitpod-dev.com/workspaces" target="_blank">hw-papi-wa87eb04d5c1.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-papi-watch-in-service-gha.21496</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-papi-wa87eb04d5c1%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/service/service.tsx [hunk 1]
```diff
@@ -164,8 +165,9 @@ export class IDEFrontendService implements IDEFrontendDashboardService.IServer {
     private ownerId: string | undefined;
     private user: User | undefined;
     private ideCredentials!: string;
+    private workspace!: Workspace;
```
Comment (geropl) [inline components/dashboard/src/service/service.tsx:168]: I'm wondering a bit why this needs to be an instance member, as it _seems_ it's only used [here](https://github.com/gitpod-io/gitpod/pull/19213/files#diff-c8930e07c7cd1e6e259fdef732fcb303539923abb895879cd31664a5379d2d51R224) (to have one initial `Workspace` shape to fallback to. Could it be passed per context?

Or is this just the pattern we are doing here? (Not familiar with this code).

**Comment by geropl** [comment]
Testing...

**Review by geropl — APPROVED** [review]
Code LGTM, tested and works beautifully! :heavy_check_mark:

File: components/dashboard/src/service/service.tsx (hunk 1 above)
Comment (mustard-mh) [reply components/dashboard/src/service/service.tsx:168]: We use `this.workspace` inside `parseInfo` function too. Some metadata fields of workspace are required

**Comment by mustard-mh** [comment]
Thank you! @geropl

**Pull request merged**